sellercenter-seller/src/main/java/com/gfg/sellercenter/seller/SellerReaderService.java

```diff
@@ -0,0 +1,51 @@
+package com.gfg.sellercenter.seller;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.gfg.sellercenter.seller.infra.JsonHttpReader;
+import lombok.AllArgsConstructor;
+import org.json.JSONArray;
+
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@AllArgsConstructor
+public class SellerReaderService implements SellerReaderServiceInterface {
+    @NotNull
+    @NotEmpty
+    private final String hostUrl;
+
+    @NotNull
+    private final JsonHttpReader jsonReader;
```

Comment:
I already an issue with name collisions of this HTTP reader in other libs, I think it makes sense if we encapsulate reader creation with the factory and hide this dependency completely from outside.

Reply:
Sure, let's unblock Igor and do it here https://gfgroup.atlassian.net/browse/TML-31553

Reply:
Actually, we don't need JsonHttpReader at all, mapper can do everything out of the box. Please take a look now.

Reply:
One more thing I did is removed throwing a separate exception when JsonProcessingException occurs. This exception is a child class of IOException which is also thrown there and I think that clients does not care what exactly was wrong (timeout, request rejected, or string "500 Server Error" instead of valid JSON). In any case result is the same: we can't return objects due to some problems not on client side.